statsmodels/regression/special_linear_model.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jan 16 22:31:14 2016
+
+Author: Josef Perktold
+License: BSD-3
+"""
+
+import numpy as np
+from scipy import sparse
+#import scipy.sparse.linalg as sparsela
+import pandas as pd
+
+
+from statsmodels.regression.linear_model import OLS, WLS
+from statsmodels.tools._sparse import PartialingSparse, dummy_sparse
+
+def cat2dummy_sparse(xcat, use_pandas=False):
+    """categorical to sparse dummy, use pandas for quick implementation
+
+    xcat needs to be ndarray for now
+
+    use_pandas is currently only for testing purposes, it uses dense intermediat dummy
+    """
+    # prepare np array for column iteration
+    if xcat.ndim == 1:
+        xcat_t = [xcat]
+    else:
+        xcat_t = xcat.T
+
+    if use_pandas:
+        dfs = [pd.get_dummies(xc) for xc in xcat_t]
+        x = np.column_stack(dfs)[:, 1:]   # full rank
+        xsp = sparse.csc_matrix(x)  #.astype(int))  # int breaks in LU-solve
+    else:
+        ds = [dummy_sparse(xc) for xc in xcat_t]
+        xsp = sparse.hstack(ds, format='csr')[:, 1:]   # full rank
+
+    return xsp
+
+
+def _group_demean_iterative(exog_dense, groups, add_mean=True, max_iter=10,
+                            atol=1e-8):
+    """iteratively demean an array for two-way fixed effects
+
+    This is intended for almost balanced panels. The data is converted
+    to a 3-dimensional array with nans for missing cells.
```

Comment:
does this still require that entire data can be represented as 2_D, i.e. unique 2column multi-index?
I guess so.